utils/src/main/kotlin/storage/LocalFileStorage.kt

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.utils.storage
+
+import com.here.ort.utils.safeMkdirs
+
+import java.io.File
+import java.io.InputStream
+
+/**
+ * A [FileStorage] that stores file in a [directory] of the local file system.
+ */
+class LocalFileStorage(
+    /**
+     * The directory used to store files in.
+     */
+    val directory: File,
+
+    /**
+     * If true, try to create the [directory] if it does not exist.
+     */
+    val createDirectory: Boolean = true
+) : FileStorage {
+    init {
+        require (!directory.isFile) {
+            "Cannot use storage directory '${directory.invariantSeparatorsPath}' because it is a file."
+        }
+
+        if (createDirectory) {
+            directory.safeMkdirs()
+        } else {
+            require (directory.isDirectory) {
+                "Cannot use storage directory '${directory.invariantSeparatorsPath}' because it is not a directory."
+            }
+        }
+    }
+
+    override fun read(path: String): InputStream {
+        val file = directory.resolve(path)
+
+        require(file.canonicalFile.startsWith(directory.canonicalFile)) {
+            "Path '$path' is not in directory '${directory.invariantSeparatorsPath}'."
+        }
+
+        require(file.isFile) {
+            if (file.isDirectory) {
+                "The file at '$path' is a directory."
+            } else {
+                "The file at '$path' does not exist."
```

Comment:
It could also be a symlink (or something else that is neither a file nor a directory, like maybe a device).

Reply:
Done.